asv_bench/benchmarks/benchmarks.py

```diff
@@ -819,4 +819,185 @@ def time_index(self, shape):
+        self.s_subset = self.s[::2]
+        self.s_subset_no_cache = self.s[::2].copy()
+
+        mi = IMPL.MultiIndex.from_product([rng, range(100)])
+        self.s2 = IMPL.Series(np.random.randn(len(mi)), index=mi)
+        self.s2_subset = self.s2[::2].copy()
+        execute(self.df)
+        execute(self.df2)
+        execute(self.s)
+        execute(self.s2_subset)
+        execute(self.s2)
+        execute(self.s_subset)
+        execute(self.s_subset_no_cache)
+
+    def time_reindex_dates(self, shape):
+        execute(self.df.reindex(self.rng_subset))
+
+    def time_reindex_columns(self, shape):
+        execute(self.df2.reindex(columns=self.df.columns[1:5]))
+
+    def time_reindex_multiindex_with_cache(self, shape):
+        # MultiIndex._values gets cached (pandas specific)
+        execute(self.s.reindex(self.s_subset.index))
+
+    def time_reindex_multiindex_no_cache(self, shape):
+        # Copy to avoid MultiIndex._values getting cached (pandas specific)
+        execute(self.s.reindex(self.s_subset_no_cache.index.copy()))
+
+    def time_reindex_multiindex_no_cache_dates(self, shape):
+        # Copy to avoid MultiIndex._values getting cached (pandas specific)
+        execute(self.s2_subset.reindex(self.s2.index.copy()))
+
+
+class TimeReindexMethod:
+
+    params = [
+        get_benchmark_shapes("TimeReindexMethod"),
+        ["pad", "backfill"],
+        [IMPL.date_range, IMPL.period_range],
+    ]
+    param_names = ["shape", "method", "constructor"]
+
+    def setup(self, shape, method, constructor):
+        N = shape[0]
+        self.idx = constructor("1/1/2000", periods=N, freq="1min")
+        self.ts = IMPL.Series(np.random.randn(N), index=self.idx)[::2]
+        execute(self.ts)
+
+    def time_reindex_method(self, shape, method, constructor):
+        execute(self.ts.reindex(self.idx, method=method))
+
+
+class TimeFillnaMethodSeries:
+
+    params = [get_benchmark_shapes("TimeFillnaMethodSeries"), ["pad", "backfill"]]
+    param_names = ["shape", "method"]
+
+    def setup(self, shape, method):
+        N = shape[0]
+        self.idx = IMPL.date_range("1/1/2000", periods=N, freq="1min")
+        ts = IMPL.Series(np.random.randn(N), index=self.idx)[::2]
+        self.ts_reindexed = ts.reindex(self.idx)
+        self.ts_float32 = self.ts_reindexed.astype("float32")
+        execute(self.ts_reindexed)
+        execute(self.ts_float32)
+
+    def time_reindexed(self, shape, method):
+        execute(self.ts_reindexed.fillna(method=method))
+
+    def time_float_32(self, shape, method):
+        execute(self.ts_float32.fillna(method=method))
+
+
+class TimeFillnaMethodDataframe:
+
+    params = [get_benchmark_shapes("TimeFillnaMethodDataframe"), ["pad", "backfill"]]
+    param_names = ["shape", "method"]
+
+    def setup(self, shape, method):
+        N = shape[0]
+        self.idx = IMPL.date_range("1/1/2000", periods=N, freq="1min")
+        td = IMPL.DataFrame(np.random.randn(*shape), index=self.idx)[::2]
```

Comment:
what's mean `td`?

Reply:
In ASV pandas benchmark for fillna for series ts was used to denote data was time series. 
Had confusion regarding naming of a time series indexed dataframe. So named it `td`, would `ts_df` be a good enough variable name for a dataframe holding time series?

Reply:
I don't see a particular need to specify the index type in the variable names for `fillna` operation, let's just call it `df`

Reply:
@arunjose696 please correct the name of the variable, fix CI problems and after that the PR can be merged.